tools/statvar_importer/place/place_resolver_test.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import unittest
+from unittest.mock import patch, call
+
+_SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(_SCRIPT_DIR)
+sys.path.append(os.path.dirname(_SCRIPT_DIR))
+sys.path.append(
+    os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(_SCRIPT_DIR))),
+                 'util'))
+
+from place_resolver import PlaceResolver
+
+
+class GetLookupNameTest(unittest.TestCase):
+
+    def test_get_lookup_name(self):
+        """Tests that the lookup name is correctly formed from place name and country."""
+        resolver = PlaceResolver()
+        place = {'place_name': 'Mountain View', 'country': 'USA'}
+        self.assertEqual(resolver._get_lookup_name('key', place),
+                         'Mountain View USA')
+
+    def test_get_lookup_name_country_in_name(self):
+        """Tests that the country is not appended if it's already in the place name."""
+        resolver = PlaceResolver()
+        place = {'place_name': 'Mountain View, USA', 'country': 'USA'}
+        self.assertEqual(resolver._get_lookup_name('key', place),
+                         'Mountain View, USA')
+
+    def test_get_lookup_name_use_key(self):
+        """Tests that the key is used as the place name if 'place_name' is not present."""
+        resolver = PlaceResolver()
+        place = {'country': 'USA'}
+        self.assertEqual(resolver._get_lookup_name('Mountain View', place),
+                         'Mountain View USA')
+
+
+class ResolveNameDcApiTest(unittest.TestCase):
+
+    @patch('place_resolver.PlaceResolver.resolve_name_dc_api_batch')
+    def test_resolve_name_dc_api(self, mock_dc_api):
+        """Tests that the DC API is called and returns the correct dcids."""
+        resolver = PlaceResolver(config_dict={'dc_api_key': 'test_key'})
```

Comment:
can we also test the resolved places are written to a cache file?
adding ` 'places_resolved_csv': '<test cache file>'` to the config should persist the lookups.

Reply:
Added a test for this.